### PR TITLE
Clear Hint in Table Engine and Sources

### PR DESCRIPTION
**Pull request by jsc0218**
solve [64552](https://github.com/ClickHouse/ClickHouse/issues/64552)

### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
As in the new version, SOURCES are checked based on Table Engine logic, even grant table engine is disabled by default, if a source is not granted, a prompt of table engine would popup instead, which is misleading. 

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit e348186ba26072eb76e549cd2a0adcd801c92bc6 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65419/e348186ba26072eb76e549cd2a0adcd801c92bc6/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Access/Common/AccessRightsElement.cpp
Comment (kssenii) [inline src/Access/Common/AccessRightsElement.cpp:230]: "Without ON" -> `result + " ON {db.table}"` 🤔 ? what did you mean?

File: src/Access/ContextAccess.cpp
Comment (kssenii) [inline src/Access/ContextAccess.cpp:272]: Delete this? Looks unused and I do not see why we need it

File: src/Access/ContextAccess.cpp
Comment (kssenii) [inline src/Access/ContextAccess.cpp:625]: By our style guide it should be `new_flags`

**Review by kssenii — COMMENTED** [review]
Let's add a test that we get a correct exception message in the case described in the linked issue (or similar)

File: src/Access/ContextAccess.cpp
Comment (jsc0218) [reply src/Access/ContextAccess.cpp:272]: It is like the one just 2 lines before it, to pass the compile of template since there exists a case with no argument.

Comment (jsc0218) [reply src/Access/ContextAccess.cpp:272]: @kssenii could you take a look again, looks like the test failure is irrevalent?

**Review by kssenii — APPROVED** [review]

**Comment by jsc0218** [comment]
looks like not fixed yet [02922_deduplication_with_zero_copy](https://github.com/ClickHouse/ClickHouse/pull/65492)

**Comment by jsc0218** [comment]
blocked by [00993_system_parts_race_condition_drop_zookeeper](https://github.com/ClickHouse/ClickHouse/issues/36099)

**Comment by jsc0218** [comment]
blocked by [this](https://github.com/ClickHouse/ClickHouse/issues/57418)

**Pull request merged**
